crates/rust-eigenda-v2-client/src/errors.rs

```diff
@@ -0,0 +1,46 @@
+/// Errors returned by this crate
+#[derive(Debug, thiserror::Error)]
+pub enum EigenClientError {
+    #[error(transparent)]
+    Conversion(#[from] ConversionError),
+    #[error(transparent)]
+    Blob(#[from] BlobError),
+}
+
+/// Errors specific to conversion
+#[derive(Debug, thiserror::Error)]
+pub enum ConversionError {
+    #[error("Failed to parse payload: {0}")]
+    Payload(String),
+    #[error("Failed to parse encoded payload: {0}")]
+    EncodedPayload(String),
+    #[error("Failed to parse polynomial: {0}")]
```

Comment:
```suggestion
    #[error("Failed to convert polynomial: {0}")]
```

Reply:
This error seems to only happen in the convert functions (eval->coeff and coeff->eval), and not related to parsing.